ui/src/app/exam/editor/publication/examPublication.component.ts

```diff
@@ -144,11 +157,34 @@ export class ExamPublicationComponent implements OnInit {
 
     updateExam = () => this.updateExam$().subscribe();
 
-    setExamDuration = (duration: number) => {
+    setExamDuration = (hours: number | undefined, minutes: number | undefined) => {
+        // Fix undefined values
+        const fixHour = hours || 0;
+        const fixMinutes = minutes || 0;
+        const duration = fixHour * 60 + fixMinutes;
+        if (duration < this.minDuration || duration > this.maxDuration) {
+            toast.warning(this.translate.instant('DIALOGS_ERROR'));
+            return null;
+        }
         this.exam.duration = duration;
```

Comment:
Tää kannattaa setata vasta tuolla subscriptionin sisällä. Tälleen varmistetaan että käyttis ei päivity ennen kuin päivitys on mennyt backendille onnistuneesti läpi.

this.updateExam${}.subscribe(() => this.exam.duration = duration);